addons.conf

```diff
@@ -48,21 +44,19 @@ server {
     }
 
     location @olympia {
-        # We are using uwsgi in our dev, stage and prod environments
-        # but not yet for our local development. So let's simply do a
-        # proxy_pass here to redirect to the django runserver.
-        proxy_pass http://web;
-        proxy_set_header X-FORWARDED-PROTOCOL "ssl";
-        proxy_set_header Host $http_host;
-        proxy_redirect off;
-        proxy_set_header X-Real-IP $remote_addr;
+        uwsgi_pass  web;
+        include     uwsgi_params;
+        uwsgi_read_timeout 60;
+        uwsgi_connect_timeout 60;
 
-        proxy_connect_timeout 30;
-        proxy_read_timeout 30;
+        uwsgi_param   Host                  $http_host;
+        uwsgi_param   X-Real-IP             $remote_addr;
+        uwsgi_param   X-Forwarded-For       $proxy_add_x_forwarded_for;
+        uwsgi_param   X-Forwarded-Protocol  ssl;
+        uwsgi_param   Vary                  User-Agent;
```

Comment:
Also in production, we don't pass the "Vary" header to the app, we instead `add_header Vary User-Agent;`, which basically add a "Vary" header when Nginx replies to the request.


You can see https://github.com/mozilla-services/cloudops-deployment/blob/master/projects/amo/puppet/modules/amo_proxy/templates/nginx.addons.conf.erb#L516-L533 for what we do for the addons-server.

Reply:
Indeed, fixed in 7f2473d